app/modules/pin/pin.controller.ts

```diff
@@ -0,0 +1,35 @@
+import { InternalServerError } from 'restify-errors';
+import IController from '../../interfaces/utils/IController';
+import { IRequest, IResponse } from '../../interfaces/utils/IServer';
+import Pin from '../../models/pin.model';
+import * as restify from 'restify';
+
+export default class PeriodicalCronListener implements IController {
+  public async post(req: IRequest, res: IResponse, next: restify.Next) {
+    try {
+      const pin = req.body;
+
+      const saved = await Pin.create({ code: pin.code, phone: pin.phone });
```

Comment:
Check if the phone is unique and if its not, return an error to the user.

Reply:
Will do.